frontend/web/components/modals/CreateFlag.js

```diff
@@ -555,6 +590,27 @@ const CreateFlag = class extends Component {
             />
           </FormGroup>
         )}
+        {metadataEnable && featureContentType?.id && (
+          <FormGroup className='mb-5 setting'>
+            <InputGroup
+              title={'Metadata'}
+              tooltip={`${Constants.strings.TOOLTIP_METADATA_DESCRIPTION} feature`}
+              tooltipPlace='left'
+              component={
+                <AddMetadataToEntity
+                  organisationId={AccountStore.getOrganisation().id}
+                  projectId={this.props.projectId}
+                  entityId={projectFlag?.id}
```

Comment:
Hmmm, a question here. If meta is required and requires an entity id, how does this work for creating a feature?

Reply:
The logic for adding the metadata now resides within createFeature. It will be saved or created when the feature is saved or created.
The entity ID is used to retrieve the feature's metadata in case the feature already exists.